sixtyfps_runtime/corelib/graphics/image.rs

```diff
@@ -11,6 +11,175 @@ LICENSE END */
 use crate::slice::Slice;
 use crate::{SharedString, SharedVector};
 
+/// SharedPixelBuffer is a container for storing image data as pixels, backed by
+/// [`SharedVector`]. That means it is internally reference counted and cheap
+/// to clone.
+///
+/// You can construct new a new empty shared pixel buffer with [`SharedPixelBuffer::new`],
+/// or you can clone it from an existing contiguous buffer that you might already have, using
+/// [`SharedPixelBuffer::clone_from_slice`].
+///
+/// See the documentation for [`Image`] for examples how to use this type to integrate
+/// SixtyFPS with external rendering functions.
+#[derive(Debug, Clone)]
+#[repr(C)]
+pub struct SharedPixelBuffer<Pixel> {
+    width: usize,
+    height: usize,
+    stride: usize,
+    data: SharedVector<Pixel>,
+}
+
+impl<Pixel> SharedPixelBuffer<Pixel> {
+    /// Returns the width of the image in pixels.
+    pub fn width(&self) -> usize {
+        self.width
+    }
+
+    /// Returns the height of the image in pixels.
+    pub fn height(&self) -> usize {
+        self.height
+    }
+
+    /// Returns the number of pixels per line.
+    pub fn stride(&self) -> usize {
+        self.stride
+    }
+}
+
+impl<Pixel: Clone> SharedPixelBuffer<Pixel> {
+    /// Return a mutable slice to the pixel data. If the SharedPixelBuffer was shared, this will make a copy of the buffer.
+    pub fn as_mut_slice(&mut self) -> &mut [Pixel] {
```

Comment:
Wondering if make_mut_slice would not be a better name. What do you think?

Reply:
As discussed, yes ver good point, that's a better name. I'll make the corresponding change to this and SharedVector after merging.